edweiss-app/model/lectures/lectureDoc.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file lectureDoc.ts
+ * @description Model for managing lecture documents and events in the edweiss app
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import { FunctionFolder, FunctionOf } from '../functions';
+import Quizzes from '../quizzes';
+
+// ------------------------------------------------------------
+// -------------------   Lecture Namespace   ------------------
+// ------------------------------------------------------------
+
+namespace LectureDisplay {
+
+    export type LectureEvents = "quiz";
+
+    interface LectureEventBase {
+        type: LectureEvents;
+        done: boolean;
+        pageNumber: number;
+    }
+
+    export interface QuizLectureEvent extends LectureEventBase {
+        type: "quiz";
+        quizModel: Quizzes.Quiz;
+    }
+
+    export interface Lecture {
```

Comment:
Maybe audioTrranscript is an optional variable of Lecture, you can add a '?'.

Reply:
True it might indeed not be something that the teacher decides to do thank you I will adapt it right away